File: 136-Single Number/single_number.dart
main() {
  final nums = [2, 2, 1];
  SingleNumber.singleNumber(nums);
}

abstract class SingleNumber {
  static int singleNumber(List<int> nums) {
    //
    // First attempt 25 June 2024 16:06
    // Runtime: 346ms, Memory: 150.20MB
    //
    int val = 0;

    for (int i = 0; i < nums.length; i++) {
      val = val ^ nums[i];
    }

    return val;
  }
}

File: 136-Single Number/SingleNumber.kt
fun main() {
    val nums = intArrayOf(2, 2, 1)

    println("Output: ${SingleNumber.singleNumber(nums)}")
}

object SingleNumber {
    fun singleNumber(nums: IntArray): Int {
        //
        // First attempt 25 June 2024 16:02
        // Runtime: 216ms, Memory: 39.59MB
        //
        var value = 0

        for (num in nums) {
            value = value xor num
        }

        return value
    }
}